crates/bevy_render/src/color/mod.rs

```diff
@@ -406,6 +408,237 @@ impl Color {
+    #[must_use]
+    pub fn with_linear_rgb_r(mut self, r: f32) -> Self {
+        self.set_linear_rgb_r(r);
+        self
+    }
+
+    /// Set the green in linear rgb colorspace.
+    pub fn set_linear_rgb_g(&mut self, g: f32) -> &mut Self {
+        *self = self.as_rgba_linear();
+        match self {
+            Color::RgbaLinear { green, .. } => *green = g,
+            _ => unreachable!(),
+        }
+        self
+    }
+
+    /// Returns this color with green set to a new value in linear rgb colorspace.
+    #[must_use]
+    pub fn with_linear_rgb_g(mut self, g: f32) -> Self {
+        self.set_linear_rgb_g(g);
+        self
+    }
+
+    /// Set the blue in linear rgb colorspace.
+    pub fn set_linear_rgb_b(&mut self, b: f32) -> &mut Self {
+        *self = self.as_rgba_linear();
+        match self {
+            Color::RgbaLinear { blue, .. } => *blue = b,
+            _ => unreachable!(),
+        }
+        self
+    }
+
+    /// Returns this color with blue set to a new value in linear rgb colorspace.
+    #[must_use]
+    pub fn with_linear_rgb_b(mut self, b: f32) -> Self {
+        self.set_linear_rgb_b(b);
+        self
+    }
+
+    // Hsl
+
+    /// Get the hue in hsl colorspace.
+    pub fn hsl_hue(&self) -> f32 {
+        match self.as_hsla() {
+            Color::Hsla { hue, .. } => hue,
+            _ => unreachable!(),
+        }
+    }
+
+    /// Get the lightness in hsl colorspace.
+    pub fn hsl_lightness(&self) -> f32 {
+        match self.as_hsla() {
+            Color::Hsla { lightness, .. } => lightness,
+            _ => unreachable!(),
+        }
+    }
+
+    /// Get the saturation in hsl colorspace.
+    pub fn hsl_saturation(&self) -> f32 {
+        match self.as_hsla() {
+            Color::Hsla { saturation, .. } => saturation,
+            _ => unreachable!(),
+        }
+    }
+
+    /// Set the hue in hsl colorspace.
+    pub fn set_hsl_hue(&mut self, hue: f32) -> &mut Self {
+        *self = self.as_hsla();
+        match self {
+            Color::Hsla { hue: h, .. } => *h = hue,
+            _ => unreachable!(),
+        }
+        self
+    }
+
+    /// Returns this color with hue set to a new value in hsl colorspace.
+    #[must_use]
+    pub fn with_hsl_hue(mut self, hue: f32) -> Self {
+        self.set_hsl_hue(hue);
+        self
+    }
+
+    /// Set the saturation in hsl colorspace.
+    pub fn set_hsl_saturation(&mut self, saturation: f32) -> &mut Self {
+        *self = self.as_hsla();
+        match self {
+            Color::Hsla { saturation: s, .. } => *s = saturation,
+            _ => unreachable!(),
+        }
+        self
+    }
+
+    /// Returns this color with saturation set to a new value in hsl colorspace.
+    #[must_use]
+    pub fn with_hsl_saturation(mut self, saturation: f32) -> Self {
+        self.set_hsl_saturation(saturation);
+        self
+    }
+
+    /// Set the lightness in hsl colorspace.
+    pub fn set_hsl_lightness(&mut self, lightness: f32) -> &mut Self {
```

Comment:
The alpha setter method is shared across all color spaces with alpha, does it make sense to also share the lightness method across all color spaces that support lightness? (Assuming the scale is the same? If not, disregard)

Reply:
Close but not the same
10 in lch is 10.588 in hsl